packages/cli-kit/src/public/node/environments.ts

```diff
@@ -1,36 +1,27 @@
 import {decodeToml} from './toml.js'
-import {fileExists, readFile, findPathUp} from './fs.js'
-import {joinPath} from './path.js'
+import {fileExists, readFile} from './fs.js'
+import {outputWarn} from './output.js'
 import {JsonMap} from '../../private/common/json.js'
 
-export const environmentsFilename = 'shopify.environments.toml'
-
 export interface Environments {
   [name: string]: JsonMap
 }
 /**
- * Loads environments from a directory.
- * @param dir - The directory to load environments from.
- * @param opts - Options for loading environments, including:
- * - findUp: whether to search upwards for an environments file.
+ * Loads environments from a file.
+ * @param dir - The file path to load environments from.
  * @returns The loaded environments.
  */
-export async function loadEnvironmentsFromDirectory(dir: string, opts?: {findUp: boolean}): Promise<Environments> {
-  let environmentsFilePath: string | undefined
-  if (opts?.findUp) {
-    environmentsFilePath = await findPathUp(environmentsFilename, {
-      cwd: dir,
-      type: 'file',
-    })
-  } else {
-    const allowedEnvironmentsFilePath = joinPath(dir, environmentsFilename)
-    if (await fileExists(allowedEnvironmentsFilePath)) {
-      environmentsFilePath = allowedEnvironmentsFilePath
-    }
-  }
-  if (environmentsFilePath) {
-    return decodeToml(await readFile(environmentsFilePath)) as Environments
+export async function loadEnvironment(
```

Comment:
I'd design this API such that Typescript can prevent invalid states from being passed. The following invocation is possible and doesn't do anything besides calling `outputWarn`:

```js
loadEnvironment("....", undefined)
```

I'd also consider moving the `findUp` logic inside this function. If it's the command knows the name of the environment file and the path, we can pass that information into this function and do the find up from here:

```js
await loadEnvironment("filename", {
  from: directory
})
```